ui/components/GHImport.js

```diff
@@ -119,6 +128,12 @@ const GHImport = ({ onSubmit }) => {
                     <FormControl className={classes.formControl}>
                       <TextField id="outlined-basic" label="Path" variant="outlined"  onChange={() => setPath(`${path}/${ recursive ? '**' : ''}`)}/>
```

Comment:
Here, what we are trying to do is if the user wants us to recursively search then we add `**` to the directory path else we don't. And since state `check` tells us if user wants the recursive search or not, we need to use `check` here instead of `recursive`. Since you have declared recursive as `const` and had set its value as `false`, its always going to be false so the current code is not dynamic. You can make it work by using `check `: 
```onChange={() => setPath(`${path}/${ check ? '**' : ''}`)}```